### DISCUSSION TITLE
How to retrieve localized resources from `Microsoft.UI.Xaml.Common.rc`

### DISCUSSION THREAD

**Discussion by MartinZikmund** [microsoft/microsoft-ui-xaml#9049] [Q&A]
We are trying to get resources which are defined in `microsoft-ui-xaml\dxaml\xcp\dxaml\dllsrv\winrt\Microsoft.UI.Xaml.Common.rc`, but this file only seems to contain english localization. However, in a running WinAppSDK app they are definitely localized. We tried querying `ResourceManager` but the `Microsoft.WindowsAppRuntime.1.4` map does not contain these. Can you please help us understand how the localization happens and if there is some way to enumerate these resources for all supported cultures?

**Comment by DarranRowe** [comment]
The .rc extension would indicate a native Windows API resource type. If you look in the Windows App SDK package, you can see all of those directories with language names.

![Screenshot 2023-11-03 144829](https://github.com/microsoft/microsoft-ui-xaml/assets/52577874/9062a84f-28bc-4a61-af60-2ebdcac7c530)

Each of those directories then contain .mui files.

![Screenshot 2023-11-03 144840](https://github.com/microsoft/microsoft-ui-xaml/assets/52577874/651342f5-9c04-48de-8352-f78445ec1e88)

Because these are native Windows API resources, this means that you would use the Windows API resource and MUI functions to get access to these resources. The learn link to the main MUI documentation is [here](https://learn.microsoft.com/en-us/windows/win32/intl/multilingual-user-interface). The learn link for the resources documentation is [here](https://learn.microsoft.com/en-us/windows/win32/menurc/resources).
As for how the localisation occurs, the .rc file you mention contains the statement:

```
#include "localizedResource.h"
```
So that is what makes the resources visible to the resource compiler. There are tools such as the Multilingual App Toolkit that allows you provide information and it then generates the required resources in a format that the resource compiler will accept.
